activetable/activetable.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+"""An XBlock with a tabular problem type that requires students to fill in some cells."""
+from __future__ import absolute_import, division, unicode_literals
+
+import textwrap
+
+from xblock.core import XBlock
+from xblock.fields import Dict, Float, Integer, Scope, String
+from xblock.fragment import Fragment
+from xblock.validation import ValidationMessage
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from .cells import NumericCell
+from .parsers import ParseError, parse_table, parse_number_list
+
+loader = ResourceLoader(__name__)  # pylint: disable=invalid-name
+
+
+class ActiveTableXBlock(StudioEditableXBlockMixin, XBlock):
+    """An XBlock with a tabular problem type that requires students to fill in some cells."""
+
+    display_name = String(
+        display_name='Display Name',
+        help='The title Studio uses for the component.',
+        scope=Scope.settings,
+        default='ActiveTable problem'
+    )
+    table_definition = String(
```

Comment:
See the above comment about child blocks. I really think we should be using them here-- this methodology was built out of limitations that don't exist in XBlock.